### ISSUE TITLE
Queue Support

### ISSUE THREAD

**Issue by PostsDesert** [stevenleeg/geemusic#5]
It would be nice to have support to add a song to the end of the queue so you could say `next play {song_name}` or `queue {song_name}`.

**Comment by stevenleeg** [comment]
This could be a good addition but I'm a bit worried about the lack of feedback if the wrong song is added to the queue (as of now if the matched song is incorrect you immediately know and yell "Alexa stop!". Maybe we could have a reprompt, so the dialog would be something like:

> **User**: Alexa, tell GeeMusic to queue `{song_name}`.
> **Alexa:** Is `{song_name}` by `{artist_name}`?
> **User:** Yes.
> **Alexa:** Queued.

> **User**: Alexa, tell GeeMusic to queue `{song_name}`.
> **Alexa:** Is `{song_name}` by `{artist_name}`?
> **User:** No.
> **Alexa:** Cancelled.

This may be a bit too wordy and lead to bad UX. What do you think?

**Comment by PostsDesert** [comment]
Yeah, that would be kinda wordy. Would having an undo/revert function hurt the UX less?

>User: Alexa, tell GeeMusic to queue {song_name}.
>Alexa: Queued {song_name} by {artist_name}.

Then if the song was incorrectly matched the user could say:

>User: Alexa, tell GeeMusic to remove that queue.
>Alexa: Removed {song_name} by {artist_name} from the queue.

The downside to this approach is it makes the code messier because now you have two `@ask.intent` requests.

**Comment by stevenleeg** [comment]
@sandeepzgk as far as I can tell the Skills API doesn't allow for controlling the LED strip, so I don't think we'd be able to implement any sort of visual cue.

@PostsDesert That interaction seems like a reasonable approach to take! I don't think it's messy to break it up into multiple intents, as this use case is what they're there for.

**Comment by PostsDesert** [comment]
I got queue support set up {here](https://github.com/PostsDesert/geemusic). There has to be a better way to check again for a new song without using a global variable.

**Comment by PostsDesert** [comment]
Okay, so I have this working somewhat on [my fork here](https://github.com/PostsDesert/geemusic/tree/queue-support). I'm having problems coming up with a way to queue the song if `@playback_almost_finished` has already fired.

Cheers  🎉

**Comment by PostsDesert** [comment]
Moved discussion to PR #34.
